workspaces/arborist/lib/arborist/build-ideal-tree.js

```diff
@@ -1268,17 +1274,18 @@ This is a one-time fix-up, please be patient...
     // the object so it doesn't get mutated.
     // Don't bother to load the manifest for link deps, because the target
     // might be within another package that doesn't exist yet.
-    const { legacyPeerDeps } = this
+    const { installLinks, legacyPeerDeps } = this
+    const isWorkspace = this.idealTree.workspaces && this.idealTree.workspaces.has(spec.name)
 
-    // spec is a directory, link it
-    if (spec.type === 'directory') {
+    // spec is a directory, link it unless installLinks is set or it's a workspace
+    if (spec.type === 'directory' && (isWorkspace || !installLinks)) {
```

Comment:
So workspaces still explicitly don't get installed with this flag?  I *think* this is right but wanna be sure we have really thought about this.  What if we did install workspaces?  Would that make their install scripts run? Would that mean the workspaces themselves don't have `node_modules` folders?

Reply:
correct, workspaces will always be symlinks with this change. it mostly just felt weird to install them. they live within the same directory tree as the root project, so we already reify their dependencies anyway. didn't feel like there was much to be gained by packing and installing them